### PR TITLE
Add support for Service Profile Discovery

### PR DESCRIPTION
**Pull request by uppalk1**
* update swagger file to support /serviceProfile
* update controller file and unit test

### CODE REVIEW

**Comment by uppalk1** [comment]
@geoff-reid @tannoa2 @keedya

File: controllers/default_controller.py [hunk 1]
```diff
@@ -168,10 +167,58 @@ def getChassis(host=None, user=None, password=None):
...
         handle.logout()
-        return 'Forbiden', "", 403
+        return 'Forbidden', "", 403
     handle.logout()
     return data
 
+@http_body_factory
+def getServiceProfile(host=None, user=None, password=None):
+    handle = UcsHandle(host, user, password, secure=False)
+    if handle.login():
```
Comment (anhou) [inline controllers/default_controller.py]: so many nested if/else, could be refined such as using guard sentences, like
if  not handle.login():
        handle.logout()
        return 'Forbidden', "", 403

dosomething

File: swagger/swagger.yaml [hunk 2]
```diff
@@ -147,5 +147,31 @@ paths:
...
+    get:
+      tags:
+      - "default"
+      description: "Returns an array of Service Profiles"
+      operationId: "controllers.default_controller.getServiceProfile"
+      parameters:
+      - name: "host"
+        in: "query"
+        description: "USC manager host name or IP"
+        required: false
```
Comment (yyscamper) [inline swagger/swagger.yaml]: From the UcsHandle implementation (https://github.com/CiscoUcs/ucsmsdk/blob/master/ucsmsdk/ucshandle.py), I don't think the host/user/password can be optional, so why you define them as optional?
If this is a bug, please fix it, If this is purposely, please explain the reason.

File: controllers/default_controller.py [hunk 3]
```diff
@@ -168,10 +167,58 @@ def getChassis(host=None, user=None, password=None):
...
+                       ]
+        finalObjs = {}
+        for x in rootElements:
+            subElement = {}
+            try:
+                components = handle.query_classid(x["ciscoXmlName"])
+            except UcsException as e:
+                handle.logout()
+                return 'Internal Server Error', e.error_descr, 500
+            else:
```
Comment (anhou) [inline controllers/default_controller.py]: where's matched 'if'?

File: controllers/default_controller.py [hunk 4]
```diff
@@ -168,10 +167,58 @@ def getChassis(host=None, user=None, password=None):
...
+                                    logicalServer["path"] = obj.dn
+                                    logicalServer["associatedServer"] = obj.pn_dn
+                                    logicalServer["assoc_state"] = obj.assoc_state
+                                    subElement["members"].append(logicalServer)
+                            else:
+                                handle.logout()
+                                return "Couldn't fetch Logical Servers", "", 500
+                    finalObjs[x["humanReadableName"]] = subElement
+                    handle.logout()
+                else:
```
Comment (anhou) [inline controllers/default_controller.py]: guard sentence also could used here to simplify the logic

File: controllers/default_controller.py [hunk 5]
```diff
@@ -168,10 +167,58 @@ def getChassis(host=None, user=None, password=None):
...
+                if(type(components) == list):
+                    for y in components:
+                        subElement["org"] = y.level
+                        subElement["members"] = []
+                        try:
+                            lsList = handle.query_children(in_dn="org-root", class_id="lsServer")
+                        except UcsException as e:
+                            handle.logout()
+                            return 'Internal Server Error', e.error_descr, 500
+                        else:
```
Comment (anhou) [inline controllers/default_controller.py]: where's matched 'if'?

File: controllers/default_controller.py [hunk 6]
```diff
@@ -168,10 +167,58 @@ def getChassis(host=None, user=None, password=None):
...
+                                for item in lsList:
+                                    logicalServer = {}
+                                    identifier = item.dn
+                                    obj = handle.query_dn(dn=identifier)
+                                    logicalServer["name"] = obj.rn
+                                    logicalServer["path"] = obj.dn
+                                    logicalServer["associatedServer"] = obj.pn_dn
+                                    logicalServer["assoc_state"] = obj.assoc_state
+                                    subElement["members"].append(logicalServer)
+                            else:
```
Comment (anhou) [inline controllers/default_controller.py]: guard sentence also could used for this to simplify the logic

File: swagger/swagger.yaml (hunk 2 above)
Comment (uppalk1) [inline swagger/swagger.yaml]: This will be addressed as part of https://rackhd.atlassian.net/browse/RAC-4389

**Pull request merged**